src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,35 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
+        this.cache = new ConcurrentHashMap<>();
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> fillAndReturnUserData(userId), executor);
+    }
+
+    private UserData fillAndReturnUserData(String userId) {
+        try {
+            Thread.sleep(200);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Using `Thread.sleep` in an asynchronous processing method can be misleading, as it suggests blocking behavior in a non-blocking context. Moreover, catching `InterruptedException` and rethrowing it as a `RuntimeException` can disrupt the thread's interrupted status. Consider removing the sleep or if the artificial delay is necessary for simulating a long-running task, document it clearly. Additionally, use `CompletableFuture`'s `completeExceptionally` method to handle interruptions.